opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/DefaultProvisionService.java

```diff
@@ -789,6 +791,10 @@ public OnmsNode getRequisitionedNode(final String foreignSource, final String fo
 
         node.setCategories(dbCategories);
 
+        if (node.getLocation() == null || Strings.isNullOrEmpty(node.getLocation().getLocationName())) {
+            node.setLocation(m_monitoringLocationDao.getDefaultLocation());
```

Comment:
No, this is required by the NodeScan stuff. It uses this method to get the node from the requisition.